### ISSUE TITLE
Minor oddities with statusbar.show = 'in-mode'

### ISSUE THREAD

**Issue by rien333** [qutebrowser/qutebrowser#5520]
**Version info**:
qutebrowser v1.12.0
Git commit: 049d51d1d (2020-06-11 17:23:59 +0200)
Backend: QtWebEngine (Chromium 80.0.3987.163)
Qt: 5.15.0

**Does the bug happen if you start with `--temp-basedir`?**:

Yes.

**Description**

When setting the newly added `statusbar.show`  option to `'in-mode'`, two websites I frequently use show some odd behavior that wasn't there before. Thee don't seem to happen with `statusbar.show = 'always'`. So far, I haven't found any affected websites aside from the two below.  

Initially the problems seemed too minor to report, but given that there are two of them, there might be some flaw(s) in the code.

Other than these two oddities, the new option is pretty great. 

**How to reproduce**

Set `statusbar.show` to  `'in-mode'`. Then: 

## Oddity 1 ##

1. Go to [Stanford Encyclopedia of Philosophy](https://plato.stanford.edu/entries/merleau-ponty/)
2. Press <kbd>f</kbd> to start hinting. 
3. The page will scroll down a small  amount everytime you start hinting.

## Oddity 2 ##

1. Go to [Rate Your Music](https://rateyourmusic.com/artist/sarah-davachi). 
2. Search for part of the name of an artist with <kbd>/</kbd>, which should be at the top of the page. (e.g. "sarah davachi"; I frequently do this to quickly copy e.g. a complicated name of an artist) 
3. Press <kbd>v</kbd> to enter caret mode. 
4. Your search highlight will not turn into a selection, which it normally does.

**Comment by kontheocharis** [comment]
I can confirm that the second thing happens for me as well. It also happens when `statusbar.show=never`. I wonder if it is due to this new setting or the problem always existed (previously if `statusbar.hide=true`). However, I can't reproduce the first one; the page doesn't scroll at all on hinting.

**Comment by rien333** [comment]
> However, I can't reproduce the first one; the page doesn't scroll at all on hinting.

For some reason it doesn't work on top of the page. However, if I navigate (e.g.) halfway the article, it should scroll down a little when hinting. 

Additional observation: the root of the problem seems to lie in showing/hiding the statusbar (which happens during hinting). The stanford article scrolls down a little evertime I cycle between `statusbar.show` modes (i.e. `config.bind('b', 'config-cycle statusbar.show "always" "in-mode"')`).

**Comment by rien333** [comment]
> Edit: Okay tested this with the 1.12.0 release and it seems that issue 2 still there with statusbar.hide=true.

Hm, I don't recall the problem having existed for  a long  time (I use the technique I described for copying names quite often), and I update qutebrowser quite frequently. But yeah, there is definitley a chance not everything described here is related to the most recent changes to the statusbar code.

**Comment by kontheocharis** [comment]
> For some reason it doesn't work on top of the page. However, if I navigate (e.g.) halfway the article, it should scroll down a little when hinting.

Ah, I see. Upon inspection it seems that this website uses the jQuery plugin ScrollToFixed, which seems to be what is causing this. The same issue arises on this page: <http://bigspotteddog.github.io/ScrollToFixed/>. I think it has to do with the fact that showing and hiding the status bar changes the CSS viewport height and this plugin relies on the viewport height to calculate the positioning of elements (not sure exactly where the problem lies).

**Comment by rien333** [comment]
> think it has to do with the fact that showing and hiding the status bar changes the CSS viewport height and this plugin relies on the viewport

You know what, now I think this might be completely unrelated to any recent changes to the statusbar code. I'm so sorry! I probably only started seeing these strange behaviors recently, because I normally always have the statusbar shown.

**Comment by kontheocharis** [comment]
For <https://rateyourmusic.com/artist/sarah-davachi>, I think it is still a bug with qutebrowser. If you search for something, then press n a few times to cycle through the results, and then press v, then the search will turn into a caret selection as expected. However, if you try it on the first result, it doesn't work (just brings you to caret mode).

**Comment by The-Compiler** [comment]
Yeah, pretty sure this isn't caused by the new changes, which are [quite minor](https://github.com/qutebrowser/qutebrowser/pull/5498/files#diff-381db995711e731e9e965aad6fa1b95c). There are definitely some weird things going on with keyboard focus (and perhaps also viewport size) when the statusbar is shown/hidden, also see #2236.

I can't reproduce 1) though, even if scrolled to the middle of the page. @rien333 do you happen to use HighDPI scaling?

**Comment by rien333** [comment]
> I can't reproduce 1) though, even if scrolled to the middle of the page. @rien333 do you happen to use HighDPI scaling?

Yeah, it's most likely related to HiDPI. I remember there being some issues about that already, so feel free to mark this is a duplicate. Sorry.

**Comment by kontheocharis** [comment]
That makes sense; I also use HiDPI

**Comment by rien333** [comment]
I thought of a hacky work-around for doing a search and then a selection (two procedures I commonly perform together)  while have the statusbar hidden, if anyone else ever stumbles upon this problem. Simply show the status bar when seaching (pretty handy anyways, because the scroll percentage indicates somethig about the amount of search results left), and then hide it again upon selecting:

```python
config.bind('/', 'set statusbar.show "always" ;;  set-cmd-text /')
config.bind('y', 'yank selection ;; set statusbar.show "in-mode"', mode='caret')
```
